src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -1615,7 +1615,7 @@ protected virtual BoundNode VisitReturnStatementNoAdjust(BoundReturnStatement no
             VisitRvalue(node.ExpressionOpt);
 
             // byref return is also a potential write
-            if (node.RefKind != RefKind.None)
+            if (node.RefKind != RefKind.None && node.ExpressionOpt is object)
```

Comment:
> && node.ExpressionOpt is object [](start = 44, length = 32)

It is not clear why this needs an adjustment. Is it possible to to have a ```ref``` in source and end up without an expression?

Reply:
Yes, it's covered in a few tests.

```csharp
ref int Value() {
  return;
}
```

Reply:
The ```RefKind``` should be ```RefKind.None``` in this scenario, there is no ```ref``` in the return statement.

---
In reply to: [292252799](https://github.com/dotnet/roslyn/pull/36271#discussion_r292252799) [](ancestors = 292252799)

Reply:
➡️ This change was reverted now that the ref kind is taken from the return value.